main.cpp

```diff
@@ -402,3 +403,157 @@ int main(int argc, char** argv) {
 
     return 0;
 }
+else {
+    //User wants to Restore to Stock
+    //Check if Backup Exists
+    RenderMenuLoading(task_percent(task++), "Checking for Backup... (wave.rpx)");
+    PresentMenu();
+    if (!wave_state.backup_exists) {
+        //No Backup
+    printf("Failed - No Backup Exists\n");
+            while (WHBProcIsRunning()) { RenderMenuDone(MENU_DONE_NO_BACKUP); PresentMenu(); }
+            return -1;
+    }
+    RenderMenuLoading(task_percent(task++), "Checking for Backup... (nn_olv.rpl)");
+    PresentMenu();
+     std::ifstream is(nn_olv_path, std::ios::binary);
+        auto hash = rpx_hash(is);
+    if (hash.id == CURRENT_PRETENDO_NN_OLV || !hash.patch) {
```

Comment:
This file was never modified by martini in the first place (the console would be irrecoverably bricked if we did) so it doesn't make sense to hash it out, but this file is different anyway - see my comments below on when you restore it.